airflow/providers/cncf/kubernetes/sensors/spark_kubernetes.py

```diff
@@ -0,0 +1,78 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Dict, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.providers.cncf.kubernetes.hooks.kubernetes import KubernetesHook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class SparkKubernetesSensor(BaseSensorOperator):
+    """
+    Checks sparkApplication object in kubernetes cluster:
+       .. seealso::
+        For more detail about Spark Application Object have a look at the reference:
+        https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/api-docs.md#sparkapplication
+
+    :param application_name: spark Application resource name
+    :type application_name:  str
+    :param namespace: the kubernetes namespace where the sparkApplication reside in
+    :type namespace: str
+    :param kubernetes_conn_id: the connection to Kubernetes cluster
+    :type conn_id: str
+    """
+
+    template_fields = ('application_name', 'namespace')
+    INTERMEDIATE_STATES = ('SUBMITTED', 'RUNNING', 'PENDING_RERUN', '')
```

Comment:
What's the `''` state here for?

Reply:
the spark operator can give an empty state '' that called "new state" and after a second the state is filled with SUBMITTED, RUNNING ...